core/services/ocr2/plugins/ocr2keeper/evm21/upkeepstate/store.go

```diff
@@ -34,39 +44,52 @@ var (
 
 // upkeepStateRecord is a record that we save in a local cache.
 type upkeepStateRecord struct {
-	workID string
-	state  ocr2keepers.UpkeepState
-	block  uint64
+	WorkID          string
```

Comment:
Why do you need a separate type? Can you not just use the same struct that is defined in the ORM?

Reply:
We don't need a separate type. It was done as a separation of responsibilities.

Reply:
Right but now you have to do extra allocations. It would be more efficient to scan directly into the struct that you actually need

Reply:
This will optimize the read operation but will also increase the memory footprint of our cache (as we need more fields in DB).